fastlane/actions/project_name.rb

```diff
@@ -0,0 +1,71 @@
+require 'xcodeproj'
+
+module Fastlane
+  module Actions
+    class ProjectNameAction < Action
+
+      PROJECT_EXTENSION = ".xcodeproj".freeze
+
+      # This script returns the application name
+      # if there is a scheme that matches the project name.
+      # Otherwise, it assumes the project name should be specified 
+      # by the user and fails.
+
+      def self.run(params)
+        default_project_name
+      end
+
+      # Fastlane Action class required functions.
+
+      def self.is_supported?(platform)
+        true
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :project_name, optional: true, default_value: default_project_name)
+        ]
+      end
+
+      # Available options default_value helpers
+
+      # In case there is a single '.xcodeproj' in the default directory
+      # it can be automatically inferred by the script 
+      # if no parameter project is received.
+      def self.default_project
+        projects = Dir.entries('.').select { |each| File.extname(each) == PROJECT_EXTENSION }
+        if projects.length == 0
+          UI.abort_with_message! "No projects with extension %s found in root directory." % PROJECT_EXTENSION
+        end
+        if projects.length > 1
+          UI.abort_with_message! "Multiple projects with extension %s found in root directory." % PROJECT_EXTENSION
```

Comment:
Couldn't here the script show the user which ones there are and have him select? Or maybe "There are multiple projects: ..list.. . To select one, please add PROJECT_NAME value in the Appfile" or whatever.

Reply:
I don't think it's worth the effort, since in our projects that never happens. I added that to avoid picking happily the first with no validations.

Reply:
But can it happen that for some readon you have more than one .xcodeproj in the same folder? What if you are using a workspace? Aren't there more than one xcodeproj?

Reply:
It can happen, I can move/create an `.xcodeproj` to the same folder than another one, but we don't usually do that here (I've never seen that in fact).

In case we're using a workspace, we'd still have an only `.xcodeproj`, and a `.xcworkspace`. I've not tried the scripts with this scenario (not even before this change) so I cannot ensure it will work, but I can ensure this change won't break it in case it was working.